### PR TITLE
Merge Scaling For Better Results

### PR DESCRIPTION
**Pull request by stanbrub**
- Experimental tests for testing scaling with merge against equivalent fully generated data
- Added per-test factors (multipliers) for scaling up individual benchmarks; row, static, incremental
- Typical run time achieved locally for each benchmark operation is ~8 secs or more

- The most interesting class for review would be **StandardTestRunner**, which conditionally uses DH merge
- The vast majority of changed files in this PR have been changed to set scale for individual benchmark operation

### CODE REVIEW

File: src/it/java/io/deephaven/benchmark/tests/standard/where/WhereInTest.java [hunk 1]
```diff
@@ -13,12 +13,13 @@ public class WhereInTest {
...
     public void setup() {
+        runner.setRowFactor(6);
         runner.tables("source");
         var setup = """
         from deephaven.column import string_col
         where_filter = new_table([
-        	string_col("sPrefix", ['s1', 's2', 's3', 's4', 's5', 's6', 's7', 's8', 's9', 's10']),
-        	string_col("sSuffix", ['1s', '2s', '3s', '4s', '5s', '6s', '7s', '8s', '9s', '10s'])
+        	string_col("sPrefix", ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10']),
+        	string_col("sSuffix", ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'])
```
Comment (devinrsmith) [inline src/it/java/io/deephaven/benchmark/tests/standard/where/WhereInTest.java:22]: It looks like these column name descriptions no longer match the data; are the tests still valid?

File: src/it/java/io/deephaven/benchmark/tests/standard/StandardTestRunner.java [hunk 2]
```diff
@@ -213,20 +293,21 @@ void generateSourceTable() {
...
                 .generateParquet();
     }
 
     void generateRightTable() {
         supportTables.add("right");
         api.table("right").fixed()
-                .add("r_str250", "string", "s[1-250]")
-                .add("r_str640", "string", "[1-640]s")
+                .add("r_str250", "string", "[1-250]")
+                .add("r_str640", "string", "[1-640]")
```
Comment (devinrsmith) [inline src/it/java/io/deephaven/benchmark/tests/standard/StandardTestRunner.java:307]: These no longer have prefix / suffix, will that change the nature of tests against this data?

File: src/it/java/io/deephaven/benchmark/tests/standard/StandardTestRunner.java [hunk 3]
```diff
@@ -213,20 +293,21 @@ void generateSourceTable() {
                 .add("int640", "int", "[1-640]")
                 .add("int1M", "int", "[1-1000000]")
                 .add("float5", "float", "[1-5]")
-                .add("str250", "string", "s[1-250]")
-                .add("str640", "string", "[1-640]s")
-                .add("str1M", "string", "v[1-1000000]s")
+                .add("str250", "string", "[1-250]")
+                .add("str640", "string", "[1-640]")
+                .add("str1M", "string", "[1-1000000]")
```
Comment (devinrsmith) [inline src/it/java/io/deephaven/benchmark/tests/standard/StandardTestRunner.java:298]: These no longer have prefix / suffix, will that change the nature of tests against this data?

File: src/it/java/io/deephaven/benchmark/tests/standard/where/WhereNotInTest.java [hunk 4]
```diff
@@ -13,12 +13,13 @@ public class WhereNotInTest {
...
     public void setup() {
+        runner.setRowFactor(6);
         runner.tables("source");
         var setup = """
         from deephaven.column import string_col
         where_filter = new_table([
-        	string_col("sPrefix", ['s250', 's1', 's249', 's2', 's248']),
-        	string_col("sSuffix", ['250s', '1s', '249s', '2s', '248s'])
+        	string_col("sPrefix", ['250', '1', '249', '2', '248']),
+        	string_col("sSuffix", ['250', '1', '249', '2', '248'])
```
Comment (devinrsmith) [inline src/it/java/io/deephaven/benchmark/tests/standard/where/WhereNotInTest.java:22]: ditto

**Review by devinrsmith — COMMENTED** [review]
What's the total runtime of benchmarking after setting all of these factors?

I don't really know all the intricate details of how the tables get generated and stuff, but it seems like some of the string prefix / suffix stuff has changed, and that may change the nature of the tests?

File: src/it/java/io/deephaven/benchmark/tests/standard/where/WhereInTest.java (hunk 1 above)
Comment (stanbrub) [reply src/it/java/io/deephaven/benchmark/tests/standard/where/WhereInTest.java:22]: In StandardTestRunner, where the data is generated, the prefix/suffix has been removed there too.  So it should be fine.

File: src/it/java/io/deephaven/benchmark/tests/standard/StandardTestRunner.java (hunk 2 above)
Comment (stanbrub) [reply src/it/java/io/deephaven/benchmark/tests/standard/StandardTestRunner.java:307]: The nature of the tests is essential the same.  Still matching strings where needed, just slightly smaller ones.  (Seemed like a waste of memory for no good reason.)

File: src/it/java/io/deephaven/benchmark/tests/standard/StandardTestRunner.java (hunk 3 above)
Comment (stanbrub) [reply src/it/java/io/deephaven/benchmark/tests/standard/StandardTestRunner.java:298]: The only reason I had the prefix/suffix (e.g. "s") to begin with was to make strings more obvious in a table.  But it looks like that's color-coded in DH anyway.

File: src/it/java/io/deephaven/benchmark/tests/standard/where/WhereNotInTest.java (hunk 4 above)
Comment (stanbrub) [reply src/it/java/io/deephaven/benchmark/tests/standard/where/WhereNotInTest.java:22]: same

**Comment by stanbrub** [comment]
SelectTest was failing against 0.24.0 because of https://github.com/deephaven/deephaven-core/pull/3806. Updated "Release" workflow to use 0.24.1 instead.

**Pull request merged**